tests/spec/core/dfn-spec.js

```diff
@@ -30,17 +30,28 @@ describe("Core — Definitions", () => {
           <a>partial inner code</a>
         </section>`,
     };
+
+    const getNodeArray = NodeList => Array.prototype.slice.call(NodeList);
+
     const doc = await makeRSDoc(ops);
     const $sec = $("#dfn", doc);
-    expect($sec.find("a:contains('outerCode')").contents()[0].nodeName).toEqual(
-      "CODE"
-    );
-    expect($sec.find("a:contains('outerPre')").contents()[0].nodeName).toEqual(
-      "CODE"
-    );
-    expect($sec.find("a:contains('innerCode')").contents()[0].nodeName).toEqual(
-      "CODE"
-    );
+    const sec = doc.querySelector("#dfn");
+    const findAnchorChild = (
```

Comment:
This function looks like an overkill. We know where the anchors are, so we probably can just do:

```js
const anchors = [...sec.children].filter(node => node.localName === "a")
```

Reply:
actually this function is a replacement to $("a:contains[abcd]"). It basically finds all the child nodes of anchor tags that contain the given string and not just the anchor.

Reply:
That's what the code does, but we use it only to select:

```html
<a>outerCode</a>
<a>outerPre</a>
<a>innerCode</a>
<a>partial inner code</a>
```

which are all direct descendants of `sec`.

Reply:
I do think you are right, we could reduce this function further, the best I can come up with is something like
```javascript
[...document.querySelectorAll("a")].reduce((result, anchor) => [...result, ...anchor.childNodes], []).filter(node => node.textContent.includes("Skip"))
```

Reply:
I think we can just do anchors[0], anchors[1] rather than searching texts as we know the exact order 😁

Reply:
Nope, we want the nodes inside the anchor tags after the definitions and all the tags have been applied and not the anchor tags themselves, 
```html
<a href="#dfn-outercode" class="internalDFN" data-link-type="dfn"><code>outerCode</code></a>
<a href="#dfn-outerpre" class="internalDFN" data-link-type="dfn"><code>outerPre</code></a>
```
In this case $('a:contains["outerCode"]) returns the <code> element and not <a> itself. We must look in the child nodes